// @adyen/api-library/lib/src/typings/relayedAuthorizationWebhooks/authorisationDecisionResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthorisationDecisionResponse = void 0;
class AuthorisationDecisionResponse {
    static getAttributeTypeMap() {
        return AuthorisationDecisionResponse.attributeTypeMap;
    }
    constructor() {
    }
}
exports.AuthorisationDecisionResponse = AuthorisationDecisionResponse;
AuthorisationDecisionResponse.discriminator = undefined;
AuthorisationDecisionResponse.mapping = undefined;
AuthorisationDecisionResponse.attributeTypeMap = [
    {
        "name": "refusalReason",
        "baseName": "refusalReason",
        "type": "string",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "AuthorisationDecisionResponse.StatusEnum",
        "format": ""
    }
];
(function (AuthorisationDecisionResponse) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Authorised"] = "Authorised";
        StatusEnum["Refused"] = "Refused";
    })(StatusEnum = AuthorisationDecisionResponse.StatusEnum || (AuthorisationDecisionResponse.StatusEnum = {}));
})(AuthorisationDecisionResponse = exports.AuthorisationDecisionResponse || (exports.AuthorisationDecisionResponse = {}));
