// @adyen/api-library/lib/src/typings/legalEntityManagement/organization.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Organization = void 0;
class Organization {
    static getAttributeTypeMap() {
        return Organization.attributeTypeMap;
    }
    constructor() {
    }
}
exports.Organization = Organization;
Organization.discriminator = undefined;
Organization.mapping = undefined;
Organization.attributeTypeMap = [
    {
        "name": "countryOfGoverningLaw",
        "baseName": "countryOfGoverningLaw",
        "type": "string",
        "format": ""
    },
    {
        "name": "dateOfIncorporation",
        "baseName": "dateOfIncorporation",
        "type": "string",
        "format": ""
    },
    {
        "name": "dateOfInitiationOfLegalProceeding",
        "baseName": "dateOfInitiationOfLegalProceeding",
        "type": "string",
        "format": ""
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "doingBusinessAs",
        "baseName": "doingBusinessAs",
        "type": "string",
        "format": ""
    },
    {
        "name": "doingBusinessAsAbsent",
        "baseName": "doingBusinessAsAbsent",
        "type": "boolean | null",
        "format": ""
    },
    {
        "name": "economicSector",
        "baseName": "economicSector",
        "type": "string",
        "format": ""
    },
    {
        "name": "email",
        "baseName": "email",
        "type": "string",
        "format": ""
    },
    {
        "name": "financialReports",
        "baseName": "financialReports",
        "type": "Array<FinancialReport>",
        "format": ""
    },
    {
        "name": "globalLegalEntityIdentifier",
        "baseName": "globalLegalEntityIdentifier",
        "type": "string",
        "format": ""
    },
    {
        "name": "headOfficeIndicator",
        "baseName": "headOfficeIndicator",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "institutionalSector",
        "baseName": "institutionalSector",
        "type": "Organization.InstitutionalSectorEnum",
        "format": ""
    },
    {
        "name": "legalForm",
        "baseName": "legalForm",
        "type": "string",
        "format": ""
    },
    {
        "name": "legalName",
        "baseName": "legalName",
        "type": "string",
        "format": ""
    },
    {
        "name": "phone",
        "baseName": "phone",
        "type": "PhoneNumber | null",
        "format": ""
    },
    {
        "name": "principalPlaceOfBusiness",
        "baseName": "principalPlaceOfBusiness",
        "type": "Address | null",
        "format": ""
    },
    {
        "name": "registeredAddress",
        "baseName": "registeredAddress",
        "type": "Address",
        "format": ""
    },
    {
        "name": "registrationNumber",
        "baseName": "registrationNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "registrationNumberAbsent",
        "baseName": "registrationNumberAbsent",
        "type": "boolean | null",
        "format": ""
    },
    {
        "name": "statusOfLegalProceeding",
        "baseName": "statusOfLegalProceeding",
        "type": "Organization.StatusOfLegalProceedingEnum",
        "format": ""
    },
    {
        "name": "stockData",
        "baseName": "stockData",
        "type": "StockData | null",
        "format": ""
    },
    {
        "name": "support",
        "baseName": "support",
        "type": "Support | null",
        "format": ""
    },
    {
        "name": "taxInformation",
        "baseName": "taxInformation",
        "type": "Array<TaxInformation>",
        "format": ""
    },
    {
        "name": "taxReportingClassification",
        "baseName": "taxReportingClassification",
        "type": "TaxReportingClassification | null",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "Organization.TypeEnum",
        "format": ""
    },
    {
        "name": "vatAbsenceReason",
        "baseName": "vatAbsenceReason",
        "type": "Organization.VatAbsenceReasonEnum",
        "format": ""
    },
    {
        "name": "vatNumber",
        "baseName": "vatNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "webData",
        "baseName": "webData",
        "type": "WebData | null",
        "format": ""
    }
];
(function (Organization) {
    let InstitutionalSectorEnum;
    (function (InstitutionalSectorEnum) {
        InstitutionalSectorEnum["NonFinancialCorporation"] = "nonFinancialCorporation";
        InstitutionalSectorEnum["CentralBank"] = "centralBank";
        InstitutionalSectorEnum["CreditInstitutions"] = "creditInstitutions";
        InstitutionalSectorEnum["DepositTakingCorporations"] = "depositTakingCorporations";
        InstitutionalSectorEnum["MoneyMarketFunds"] = "moneyMarketFunds";
        InstitutionalSectorEnum["NonMmfInvestmentFunds"] = "nonMMFInvestmentFunds";
        InstitutionalSectorEnum["FinancialVehicleCorporation"] = "financialVehicleCorporation";
        InstitutionalSectorEnum["OtherFinancialIntermediaries"] = "otherFinancialIntermediaries";
        InstitutionalSectorEnum["FinancialAuxiliaries"] = "financialAuxiliaries";
        InstitutionalSectorEnum["CaptiveFinancialInstitutionsAndMoneyLenders"] = "captiveFinancialInstitutionsAndMoneyLenders";
        InstitutionalSectorEnum["InsuranceCorporations"] = "insuranceCorporations";
        InstitutionalSectorEnum["PensionFunds"] = "pensionFunds";
        InstitutionalSectorEnum["CentralGovernment"] = "centralGovernment";
        InstitutionalSectorEnum["StateGovernment"] = "stateGovernment";
        InstitutionalSectorEnum["LocalGovernment"] = "localGovernment";
        InstitutionalSectorEnum["SocialSecurityFunds"] = "socialSecurityFunds";
        InstitutionalSectorEnum["NonProfitInstitutionsServingHouseholds"] = "nonProfitInstitutionsServingHouseholds";
    })(InstitutionalSectorEnum = Organization.InstitutionalSectorEnum || (Organization.InstitutionalSectorEnum = {}));
    let StatusOfLegalProceedingEnum;
    (function (StatusOfLegalProceedingEnum) {
        StatusOfLegalProceedingEnum["NoLegalActionsTaken"] = "noLegalActionsTaken";
        StatusOfLegalProceedingEnum["UnderJudicialAdministration"] = "underJudicialAdministration";
        StatusOfLegalProceedingEnum["BankruptcyInsolvency"] = "bankruptcyInsolvency";
        StatusOfLegalProceedingEnum["OtherLegalMeasures"] = "otherLegalMeasures";
    })(StatusOfLegalProceedingEnum = Organization.StatusOfLegalProceedingEnum || (Organization.StatusOfLegalProceedingEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["AssociationIncorporated"] = "associationIncorporated";
        TypeEnum["GovernmentalOrganization"] = "governmentalOrganization";
        TypeEnum["ListedPublicCompany"] = "listedPublicCompany";
        TypeEnum["NonProfit"] = "nonProfit";
        TypeEnum["PartnershipIncorporated"] = "partnershipIncorporated";
        TypeEnum["PrivateCompany"] = "privateCompany";
    })(TypeEnum = Organization.TypeEnum || (Organization.TypeEnum = {}));
    let VatAbsenceReasonEnum;
    (function (VatAbsenceReasonEnum) {
        VatAbsenceReasonEnum["IndustryExemption"] = "industryExemption";
        VatAbsenceReasonEnum["BelowTaxThreshold"] = "belowTaxThreshold";
    })(VatAbsenceReasonEnum = Organization.VatAbsenceReasonEnum || (Organization.VatAbsenceReasonEnum = {}));
})(Organization = exports.Organization || (exports.Organization = {}));
