// @adyen/api-library/lib/src/typings/legalEntityManagement/sourceOfFunds.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SourceOfFunds = void 0;
class SourceOfFunds {
    static getAttributeTypeMap() {
        return SourceOfFunds.attributeTypeMap;
    }
    constructor() {
    }
}
exports.SourceOfFunds = SourceOfFunds;
SourceOfFunds.discriminator = undefined;
SourceOfFunds.mapping = undefined;
SourceOfFunds.attributeTypeMap = [
    {
        "name": "adyenProcessedFunds",
        "baseName": "adyenProcessedFunds",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount | null",
        "format": ""
    },
    {
        "name": "assetMonthsHeld",
        "baseName": "assetMonthsHeld",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "cryptocurrencyExchange",
        "baseName": "cryptocurrencyExchange",
        "type": "string",
        "format": ""
    },
    {
        "name": "dateOfFundsReceived",
        "baseName": "dateOfFundsReceived",
        "type": "string",
        "format": "date"
    },
    {
        "name": "dateOfSourceEvent",
        "baseName": "dateOfSourceEvent",
        "type": "string",
        "format": "date"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "financiers",
        "baseName": "financiers",
        "type": "Array<Financier>",
        "format": ""
    },
    {
        "name": "originatorLegalEntityId",
        "baseName": "originatorLegalEntityId",
        "type": "string",
        "format": ""
    },
    {
        "name": "purpose",
        "baseName": "purpose",
        "type": "string",
        "format": ""
    },
    {
        "name": "relationship",
        "baseName": "relationship",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "SourceOfFunds.TypeEnum",
        "format": ""
    },
    {
        "name": "website",
        "baseName": "website",
        "type": "string",
        "format": ""
    }
];
(function (SourceOfFunds) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Business"] = "business";
        TypeEnum["Employment"] = "employment";
        TypeEnum["Donations"] = "donations";
        TypeEnum["Inheritance"] = "inheritance";
        TypeEnum["FinancialAid"] = "financialAid";
        TypeEnum["RentalIncome"] = "rentalIncome";
        TypeEnum["DividendIncome"] = "dividendIncome";
        TypeEnum["RoyaltyIncome"] = "royaltyIncome";
        TypeEnum["ThirdPartyFunding"] = "thirdPartyFunding";
        TypeEnum["PensionIncome"] = "pensionIncome";
        TypeEnum["InsuranceSettlement"] = "insuranceSettlement";
        TypeEnum["CryptocurrencyIncome"] = "cryptocurrencyIncome";
        TypeEnum["AssetSale"] = "assetSale";
        TypeEnum["Loans"] = "loans";
        TypeEnum["GamblingWinnings"] = "gamblingWinnings";
    })(TypeEnum = SourceOfFunds.TypeEnum || (SourceOfFunds.TypeEnum = {}));
})(SourceOfFunds = exports.SourceOfFunds || (exports.SourceOfFunds = {}));
