// @adyen/api-library/lib/src/index.js
"use strict";
/*
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 * Adyen NodeJS API Library
 * Copyright (c) 2021 Adyen B.V.
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more info.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Types = exports.HttpClientException = exports.HttpURLConnectionClient = exports.hmacValidator = exports.RegionEnum = exports.EnvironmentEnum = exports.Config = exports.Client = void 0;
// For side effects
require("./typings");
// Export the main client for making API requests
var client_1 = require("./client");
Object.defineProperty(exports, "Client", { enumerable: true, get: function () { return __importDefault(client_1).default; } });
// Export the configuration class for the client
var config_1 = require("./config");
Object.defineProperty(exports, "Config", { enumerable: true, get: function () { return __importDefault(config_1).default; } });
// Export enums for different Adyen environments and regions
var config_2 = require("./config");
Object.defineProperty(exports, "EnvironmentEnum", { enumerable: true, get: function () { return config_2.EnvironmentEnum; } });
Object.defineProperty(exports, "RegionEnum", { enumerable: true, get: function () { return config_2.RegionEnum; } });
// Export all services
__exportStar(require("./services"), exports);
// Export all webhook handlers
__exportStar(require("./webhooks"), exports);
// Export a utility function for HMAC validation
var utils_1 = require("./utils");
Object.defineProperty(exports, "hmacValidator", { enumerable: true, get: function () { return utils_1.hmacValidator; } });
// Export the HTTP client implementation
var httpURLConnectionClient_1 = require("./httpClient/httpURLConnectionClient");
Object.defineProperty(exports, "HttpURLConnectionClient", { enumerable: true, get: function () { return __importDefault(httpURLConnectionClient_1).default; } });
// Export a custom exception for HTTP client errors
var httpClientException_1 = require("./httpClient/httpClientException");
Object.defineProperty(exports, "HttpClientException", { enumerable: true, get: function () { return __importDefault(httpClientException_1).default; } });
// Export all the typings under the `Types` namespace
exports.Types = __importStar(require("./typings"));
