// @adyen/api-library/lib/src/typings/transfers/fundingInstrument.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FundingInstrument = void 0;
class FundingInstrument {
    static getAttributeTypeMap() {
        return FundingInstrument.attributeTypeMap;
    }
    constructor() {
    }
}
exports.FundingInstrument = FundingInstrument;
FundingInstrument.discriminator = undefined;
FundingInstrument.mapping = undefined;
FundingInstrument.attributeTypeMap = [
    {
        "name": "cardIdentification",
        "baseName": "cardIdentification",
        "type": "CardIdentification | null",
        "format": ""
    },
    {
        "name": "networkPaymentReference",
        "baseName": "networkPaymentReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "sourceOfFunds",
        "baseName": "sourceOfFunds",
        "type": "FundingInstrument.SourceOfFundsEnum",
        "format": ""
    }
];
(function (FundingInstrument) {
    let SourceOfFundsEnum;
    (function (SourceOfFundsEnum) {
        SourceOfFundsEnum["Debit"] = "DEBIT";
        SourceOfFundsEnum["DepositAccount"] = "DEPOSIT_ACCOUNT";
    })(SourceOfFundsEnum = FundingInstrument.SourceOfFundsEnum || (FundingInstrument.SourceOfFundsEnum = {}));
})(FundingInstrument = exports.FundingInstrument || (exports.FundingInstrument = {}));
