// @adyen/api-library/lib/src/typings/checkout/checkoutOutgoingForwardRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutOutgoingForwardRequest = void 0;
class CheckoutOutgoingForwardRequest {
    static getAttributeTypeMap() {
        return CheckoutOutgoingForwardRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.CheckoutOutgoingForwardRequest = CheckoutOutgoingForwardRequest;
CheckoutOutgoingForwardRequest.discriminator = undefined;
CheckoutOutgoingForwardRequest.mapping = undefined;
CheckoutOutgoingForwardRequest.attributeTypeMap = [
    {
        "name": "body",
        "baseName": "body",
        "type": "string",
        "format": ""
    },
    {
        "name": "credentials",
        "baseName": "credentials",
        "type": "string",
        "format": ""
    },
    {
        "name": "headers",
        "baseName": "headers",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "httpMethod",
        "baseName": "httpMethod",
        "type": "CheckoutOutgoingForwardRequest.HttpMethodEnum",
        "format": ""
    },
    {
        "name": "urlSuffix",
        "baseName": "urlSuffix",
        "type": "string",
        "format": ""
    }
];
(function (CheckoutOutgoingForwardRequest) {
    let HttpMethodEnum;
    (function (HttpMethodEnum) {
        HttpMethodEnum["Post"] = "post";
        HttpMethodEnum["Put"] = "put";
        HttpMethodEnum["Patch"] = "patch";
    })(HttpMethodEnum = CheckoutOutgoingForwardRequest.HttpMethodEnum || (CheckoutOutgoingForwardRequest.HttpMethodEnum = {}));
})(CheckoutOutgoingForwardRequest = exports.CheckoutOutgoingForwardRequest || (exports.CheckoutOutgoingForwardRequest = {}));
