// @adyen/api-library/lib/src/typings/checkout/externalTokenDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExternalTokenDetails = void 0;
class ExternalTokenDetails {
    static getAttributeTypeMap() {
        return ExternalTokenDetails.attributeTypeMap;
    }
    constructor() {
    }
}
exports.ExternalTokenDetails = ExternalTokenDetails;
ExternalTokenDetails.discriminator = undefined;
ExternalTokenDetails.mapping = undefined;
ExternalTokenDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "expiryMonth",
        "baseName": "expiryMonth",
        "type": "string",
        "format": ""
    },
    {
        "name": "expiryYear",
        "baseName": "expiryYear",
        "type": "string",
        "format": ""
    },
    {
        "name": "holderName",
        "baseName": "holderName",
        "type": "string",
        "format": ""
    },
    {
        "name": "number",
        "baseName": "number",
        "type": "string",
        "format": ""
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string",
        "format": ""
    },
    {
        "name": "subtype",
        "baseName": "subtype",
        "type": "ExternalTokenDetails.SubtypeEnum",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "ExternalTokenDetails.TypeEnum",
        "format": ""
    }
];
(function (ExternalTokenDetails) {
    let SubtypeEnum;
    (function (SubtypeEnum) {
        SubtypeEnum["Hilton"] = "hilton";
    })(SubtypeEnum = ExternalTokenDetails.SubtypeEnum || (ExternalTokenDetails.SubtypeEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["ExternalToken"] = "externalToken";
    })(TypeEnum = ExternalTokenDetails.TypeEnum || (ExternalTokenDetails.TypeEnum = {}));
})(ExternalTokenDetails = exports.ExternalTokenDetails || (exports.ExternalTokenDetails = {}));
