// @adyen/api-library/lib/src/typings/checkout/paymentValidationsNameResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentValidationsNameResponse = void 0;
class PaymentValidationsNameResponse {
    static getAttributeTypeMap() {
        return PaymentValidationsNameResponse.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PaymentValidationsNameResponse = PaymentValidationsNameResponse;
PaymentValidationsNameResponse.discriminator = undefined;
PaymentValidationsNameResponse.mapping = undefined;
PaymentValidationsNameResponse.attributeTypeMap = [
    {
        "name": "rawResponse",
        "baseName": "rawResponse",
        "type": "PaymentValidationsNameResultRawResponse | null",
        "format": ""
    },
    {
        "name": "result",
        "baseName": "result",
        "type": "PaymentValidationsNameResultResponse | null",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "PaymentValidationsNameResponse.StatusEnum",
        "format": ""
    }
];
(function (PaymentValidationsNameResponse) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["NotPerformed"] = "notPerformed";
        StatusEnum["NotSupported"] = "notSupported";
        StatusEnum["Performed"] = "performed";
    })(StatusEnum = PaymentValidationsNameResponse.StatusEnum || (PaymentValidationsNameResponse.StatusEnum = {}));
})(PaymentValidationsNameResponse = exports.PaymentValidationsNameResponse || (exports.PaymentValidationsNameResponse = {}));
