// @adyen/api-library/lib/src/typings/checkout/storedPaymentMethodDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredPaymentMethodDetails = void 0;
class StoredPaymentMethodDetails {
    static getAttributeTypeMap() {
        return StoredPaymentMethodDetails.attributeTypeMap;
    }
    constructor() {
    }
}
exports.StoredPaymentMethodDetails = StoredPaymentMethodDetails;
StoredPaymentMethodDetails.discriminator = undefined;
StoredPaymentMethodDetails.mapping = undefined;
StoredPaymentMethodDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "sdkData",
        "baseName": "sdkData",
        "type": "string",
        "format": ""
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "StoredPaymentMethodDetails.TypeEnum",
        "format": ""
    }
];
(function (StoredPaymentMethodDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["AlipayPlus"] = "alipay_plus";
        TypeEnum["AlipayPlusAlipayCn"] = "alipay_plus_alipay_cn";
        TypeEnum["AlipayPlusAlipayHk"] = "alipay_plus_alipay_hk";
        TypeEnum["AlipayPlusDana"] = "alipay_plus_dana";
        TypeEnum["AlipayPlusGcash"] = "alipay_plus_gcash";
        TypeEnum["AlipayPlusKakaopay"] = "alipay_plus_kakaopay";
        TypeEnum["AlipayPlusKplus"] = "alipay_plus_kplus";
        TypeEnum["AlipayPlusNaverpay"] = "alipay_plus_naverpay";
        TypeEnum["AlipayPlusRabbitlinepay"] = "alipay_plus_rabbitlinepay";
        TypeEnum["AlipayPlusTosspay"] = "alipay_plus_tosspay";
        TypeEnum["AlipayPlusTouchngo"] = "alipay_plus_touchngo";
        TypeEnum["AlipayPlusTruemoney"] = "alipay_plus_truemoney";
        TypeEnum["BcmcMobile"] = "bcmc_mobile";
        TypeEnum["BcmcMobileQr"] = "bcmc_mobile_QR";
        TypeEnum["BcmcMobileApp"] = "bcmc_mobile_app";
        TypeEnum["MomoWallet"] = "momo_wallet";
        TypeEnum["MomoWalletApp"] = "momo_wallet_app";
        TypeEnum["PaymayaWallet"] = "paymaya_wallet";
        TypeEnum["GrabpaySg"] = "grabpay_SG";
        TypeEnum["GrabpayMy"] = "grabpay_MY";
        TypeEnum["GrabpayTh"] = "grabpay_TH";
        TypeEnum["GrabpayId"] = "grabpay_ID";
        TypeEnum["GrabpayVn"] = "grabpay_VN";
        TypeEnum["GrabpayPh"] = "grabpay_PH";
        TypeEnum["Oxxo"] = "oxxo";
        TypeEnum["Gcash"] = "gcash";
        TypeEnum["Dana"] = "dana";
        TypeEnum["Kakaopay"] = "kakaopay";
        TypeEnum["Truemoney"] = "truemoney";
        TypeEnum["Paysafecard"] = "paysafecard";
    })(TypeEnum = StoredPaymentMethodDetails.TypeEnum || (StoredPaymentMethodDetails.TypeEnum = {}));
})(StoredPaymentMethodDetails = exports.StoredPaymentMethodDetails || (exports.StoredPaymentMethodDetails = {}));
