// @adyen/api-library/lib/src/typings/checkout/tokenMandate.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenMandate = void 0;
class TokenMandate {
    static getAttributeTypeMap() {
        return TokenMandate.attributeTypeMap;
    }
    constructor() {
    }
}
exports.TokenMandate = TokenMandate;
TokenMandate.discriminator = undefined;
TokenMandate.mapping = undefined;
TokenMandate.attributeTypeMap = [
    {
        "name": "accountIdType",
        "baseName": "accountIdType",
        "type": "string",
        "format": ""
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "string",
        "format": ""
    },
    {
        "name": "amountRule",
        "baseName": "amountRule",
        "type": "TokenMandate.AmountRuleEnum",
        "format": ""
    },
    {
        "name": "billingAttemptsRule",
        "baseName": "billingAttemptsRule",
        "type": "TokenMandate.BillingAttemptsRuleEnum",
        "format": ""
    },
    {
        "name": "billingDay",
        "baseName": "billingDay",
        "type": "string",
        "format": ""
    },
    {
        "name": "count",
        "baseName": "count",
        "type": "string",
        "format": ""
    },
    {
        "name": "currency",
        "baseName": "currency",
        "type": "string",
        "format": ""
    },
    {
        "name": "endsAt",
        "baseName": "endsAt",
        "type": "string",
        "format": ""
    },
    {
        "name": "frequency",
        "baseName": "frequency",
        "type": "TokenMandate.FrequencyEnum",
        "format": ""
    },
    {
        "name": "mandateId",
        "baseName": "mandateId",
        "type": "string",
        "format": ""
    },
    {
        "name": "maskedAccountId",
        "baseName": "maskedAccountId",
        "type": "string",
        "format": ""
    },
    {
        "name": "providerId",
        "baseName": "providerId",
        "type": "string",
        "format": ""
    },
    {
        "name": "remarks",
        "baseName": "remarks",
        "type": "string",
        "format": ""
    },
    {
        "name": "startsAt",
        "baseName": "startsAt",
        "type": "string",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "string",
        "format": ""
    },
    {
        "name": "txVariant",
        "baseName": "txVariant",
        "type": "string",
        "format": ""
    }
];
(function (TokenMandate) {
    let AmountRuleEnum;
    (function (AmountRuleEnum) {
        AmountRuleEnum["Max"] = "max";
        AmountRuleEnum["Exact"] = "exact";
    })(AmountRuleEnum = TokenMandate.AmountRuleEnum || (TokenMandate.AmountRuleEnum = {}));
    let BillingAttemptsRuleEnum;
    (function (BillingAttemptsRuleEnum) {
        BillingAttemptsRuleEnum["On"] = "on";
        BillingAttemptsRuleEnum["Before"] = "before";
        BillingAttemptsRuleEnum["After"] = "after";
    })(BillingAttemptsRuleEnum = TokenMandate.BillingAttemptsRuleEnum || (TokenMandate.BillingAttemptsRuleEnum = {}));
    let FrequencyEnum;
    (function (FrequencyEnum) {
        FrequencyEnum["Adhoc"] = "adhoc";
        FrequencyEnum["Daily"] = "daily";
        FrequencyEnum["Weekly"] = "weekly";
        FrequencyEnum["BiWeekly"] = "biWeekly";
        FrequencyEnum["Monthly"] = "monthly";
        FrequencyEnum["Quarterly"] = "quarterly";
        FrequencyEnum["HalfYearly"] = "halfYearly";
        FrequencyEnum["Yearly"] = "yearly";
    })(FrequencyEnum = TokenMandate.FrequencyEnum || (TokenMandate.FrequencyEnum = {}));
})(TokenMandate = exports.TokenMandate || (exports.TokenMandate = {}));
