// @adyen/api-library/lib/src/typings/openBanking/accountVerificationCountry.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountVerificationCountry = void 0;
var AccountVerificationCountry;
(function (AccountVerificationCountry) {
    AccountVerificationCountry["Se"] = "SE";
    AccountVerificationCountry["Fi"] = "FI";
    AccountVerificationCountry["No"] = "NO";
    AccountVerificationCountry["Dk"] = "DK";
    AccountVerificationCountry["Nl"] = "NL";
    AccountVerificationCountry["It"] = "IT";
    AccountVerificationCountry["De"] = "DE";
    AccountVerificationCountry["At"] = "AT";
    AccountVerificationCountry["Es"] = "ES";
    AccountVerificationCountry["Pt"] = "PT";
    AccountVerificationCountry["Fr"] = "FR";
    AccountVerificationCountry["Gb"] = "GB";
    AccountVerificationCountry["Be"] = "BE";
    AccountVerificationCountry["Lv"] = "LV";
    AccountVerificationCountry["Lt"] = "LT";
    AccountVerificationCountry["Pl"] = "PL";
    AccountVerificationCountry["Ee"] = "EE";
    AccountVerificationCountry["Ie"] = "IE";
    AccountVerificationCountry["Us"] = "US";
    AccountVerificationCountry["Ca"] = "CA";
    AccountVerificationCountry["Au"] = "AU";
})(AccountVerificationCountry = exports.AccountVerificationCountry || (exports.AccountVerificationCountry = {}));
