// @adyen/api-library/lib/src/typings/capital/additionalBankIdentificationTypes.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdditionalBankIdentificationTypes = void 0;
var AdditionalBankIdentificationTypes;
(function (AdditionalBankIdentificationTypes) {
    AdditionalBankIdentificationTypes["AuBsbCode"] = "auBsbCode";
    AdditionalBankIdentificationTypes["CaRoutingNumber"] = "caRoutingNumber";
    AdditionalBankIdentificationTypes["GbSortCode"] = "gbSortCode";
    AdditionalBankIdentificationTypes["HkBankCode"] = "hkBankCode";
    AdditionalBankIdentificationTypes["JpZenginCode"] = "jpZenginCode";
    AdditionalBankIdentificationTypes["NzBankBranchCode"] = "nzBankBranchCode";
    AdditionalBankIdentificationTypes["UsRoutingNumber"] = "usRoutingNumber";
})(AdditionalBankIdentificationTypes = exports.AdditionalBankIdentificationTypes || (exports.AdditionalBankIdentificationTypes = {}));
