// @adyen/api-library/lib/src/typings/capital/status.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Status = void 0;
class Status {
    static getAttributeTypeMap() {
        return Status.attributeTypeMap;
    }
    constructor() {
    }
}
exports.Status = Status;
Status.discriminator = undefined;
Status.mapping = undefined;
Status.attributeTypeMap = [
    {
        "name": "actions",
        "baseName": "actions",
        "type": "Array<Action>",
        "format": ""
    },
    {
        "name": "code",
        "baseName": "code",
        "type": "Status.CodeEnum",
        "format": ""
    }
];
(function (Status) {
    let CodeEnum;
    (function (CodeEnum) {
        CodeEnum["Pending"] = "Pending";
        CodeEnum["Active"] = "Active";
        CodeEnum["Repaid"] = "Repaid";
        CodeEnum["WrittenOff"] = "WrittenOff";
        CodeEnum["Failed"] = "Failed";
        CodeEnum["Revoked"] = "Revoked";
        CodeEnum["Requested"] = "Requested";
        CodeEnum["Reviewing"] = "Reviewing";
        CodeEnum["Approved"] = "Approved";
        CodeEnum["Rejected"] = "Rejected";
        CodeEnum["Cancelled"] = "Cancelled";
    })(CodeEnum = Status.CodeEnum || (Status.CodeEnum = {}));
})(Status = exports.Status || (exports.Status = {}));
