// @adyen/api-library/lib/src/typings/management/affirmInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AffirmInfo = void 0;
class AffirmInfo {
    static getAttributeTypeMap() {
        return AffirmInfo.attributeTypeMap;
    }
    constructor() {
    }
}
exports.AffirmInfo = AffirmInfo;
AffirmInfo.discriminator = undefined;
AffirmInfo.mapping = undefined;
AffirmInfo.attributeTypeMap = [
    {
        "name": "pricePlan",
        "baseName": "pricePlan",
        "type": "AffirmInfo.PricePlanEnum",
        "format": ""
    },
    {
        "name": "supportEmail",
        "baseName": "supportEmail",
        "type": "string",
        "format": ""
    }
];
(function (AffirmInfo) {
    let PricePlanEnum;
    (function (PricePlanEnum) {
        PricePlanEnum["Bronze"] = "BRONZE";
        PricePlanEnum["Silver"] = "SILVER";
        PricePlanEnum["Gold"] = "GOLD";
    })(PricePlanEnum = AffirmInfo.PricePlanEnum || (AffirmInfo.PricePlanEnum = {}));
})(AffirmInfo = exports.AffirmInfo || (exports.AffirmInfo = {}));
