// @adyen/api-library/lib/src/typings/management/givexInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GivexInfo = void 0;
class GivexInfo {
    static getAttributeTypeMap() {
        return GivexInfo.attributeTypeMap;
    }
    constructor() {
    }
}
exports.GivexInfo = GivexInfo;
GivexInfo.discriminator = undefined;
GivexInfo.mapping = undefined;
GivexInfo.attributeTypeMap = [
    {
        "name": "currencyCode",
        "baseName": "currencyCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "password",
        "baseName": "password",
        "type": "string",
        "format": ""
    },
    {
        "name": "paymentFlow",
        "baseName": "paymentFlow",
        "type": "GivexInfo.PaymentFlowEnum",
        "format": ""
    },
    {
        "name": "username",
        "baseName": "username",
        "type": "string",
        "format": ""
    }
];
(function (GivexInfo) {
    let PaymentFlowEnum;
    (function (PaymentFlowEnum) {
        PaymentFlowEnum["Ecommerce"] = "Ecommerce";
        PaymentFlowEnum["Pos"] = "POS";
    })(PaymentFlowEnum = GivexInfo.PaymentFlowEnum || (GivexInfo.PaymentFlowEnum = {}));
})(GivexInfo = exports.GivexInfo || (exports.GivexInfo = {}));
