// @adyen/api-library/lib/src/typings/management/valuelinkInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValuelinkInfo = void 0;
class ValuelinkInfo {
    static getAttributeTypeMap() {
        return ValuelinkInfo.attributeTypeMap;
    }
    constructor() {
    }
}
exports.ValuelinkInfo = ValuelinkInfo;
ValuelinkInfo.discriminator = undefined;
ValuelinkInfo.mapping = undefined;
ValuelinkInfo.attributeTypeMap = [
    {
        "name": "authorisationMid",
        "baseName": "authorisationMid",
        "type": "string",
        "format": ""
    },
    {
        "name": "pinSupport",
        "baseName": "pinSupport",
        "type": "ValuelinkInfo.PinSupportEnum",
        "format": ""
    },
    {
        "name": "submitterId",
        "baseName": "submitterId",
        "type": "string",
        "format": ""
    },
    {
        "name": "terminalId",
        "baseName": "terminalId",
        "type": "string",
        "format": ""
    }
];
(function (ValuelinkInfo) {
    let PinSupportEnum;
    (function (PinSupportEnum) {
        PinSupportEnum["Pin"] = "PIN";
        PinSupportEnum["NoPin"] = "NO PIN";
    })(PinSupportEnum = ValuelinkInfo.PinSupportEnum || (ValuelinkInfo.PinSupportEnum = {}));
})(ValuelinkInfo = exports.ValuelinkInfo || (exports.ValuelinkInfo = {}));
