// @adyen/api-library/lib/src/services/capital/grantOffersApi.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrantOffersApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const objectSerializer_1 = require("../../typings/capital/objectSerializer");
/**
 * API handler for GrantOffersApi
 */
class GrantOffersApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://balanceplatform-api-test.adyen.com/capital/v1";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Get all available grant offers
    * @param requestOptions {@link IRequest.Options }
    * @param accountHolderId {@link string } The unique identifier of the account holder for which you want to get the available grant offers.
    * @return {@link GrantOffers }
    */
    async getAllGrantOffers(accountHolderId, requestOptions) {
        const endpoint = `${this.baseUrl}/grantOffers`;
        const resource = new resource_1.default(this, endpoint);
        const hasDefinedQueryParams = accountHolderId;
        if (hasDefinedQueryParams) {
            if (!requestOptions)
                requestOptions = {};
            if (!requestOptions.params)
                requestOptions.params = {};
            if (accountHolderId)
                requestOptions.params["accountHolderId"] = accountHolderId;
        }
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "GrantOffers");
    }
    /**
    * @summary Get the details of a grant offer
    * @param id {@link string } The unique identifier of the grant offer.
    * @param requestOptions {@link IRequest.Options }
    * @return {@link GrantOffer }
    */
    async getGrantOffer(id, requestOptions) {
        const endpoint = `${this.baseUrl}/grantOffers/{id}`
            .replace("{" + "id" + "}", encodeURIComponent(String(id)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "GrantOffer");
    }
}
exports.GrantOffersApi = GrantOffersApi;
