// @adyen/api-library/lib/src/services/capital/index.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const grantAccountsApi_1 = require("./grantAccountsApi");
const grantOffersApi_1 = require("./grantOffersApi");
const grantsApi_1 = require("./grantsApi");
const service_1 = __importDefault(require("../../service"));
class CapitalAPI extends service_1.default {
    constructor(client) {
        super(client);
    }
    get GrantAccountsApi() {
        return new grantAccountsApi_1.GrantAccountsApi(this.client);
    }
    get GrantOffersApi() {
        return new grantOffersApi_1.GrantOffersApi(this.client);
    }
    get GrantsApi() {
        return new grantsApi_1.GrantsApi(this.client);
    }
}
exports.default = CapitalAPI;
