// @adyen/api-library/lib/src/services/balancePlatform/sCAAssociationManagementApi.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SCAAssociationManagementApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const objectSerializer_1 = require("../../typings/balancePlatform/objectSerializer");
/**
 * API handler for SCAAssociationManagementApi
 */
class SCAAssociationManagementApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://balanceplatform-api-test.adyen.com/bcl/v2";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Approve a pending approval association
    * @param approveAssociationRequest {@link ApproveAssociationRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link ApproveAssociationResponse }
    */
    async approveAssociation(approveAssociationRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/scaAssociations`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(approveAssociationRequest, "ApproveAssociationRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "ApproveAssociationResponse");
    }
    /**
    * @summary Get a list of devices associated with an entity
    * @param requestOptions {@link IRequest.Options }
    * @param entityType {@link ScaEntityType } (Required) The type of entity you want to retrieve a list of associations for.   Possible values: **accountHolder** or **paymentInstrument**.
    * @param entityId {@link string } (Required) The unique identifier of the entity.
    * @param pageSize {@link number } (Required) The number of items to have on a page.   Default: **5**.
    * @param pageNumber {@link number } (Required) The index of the page to retrieve. The index of the first page is **0** (zero).   Default:  **0**.
    * @return {@link ListAssociationsResponse }
    */
    async listAssociations(entityType, entityId, pageSize, pageNumber, requestOptions) {
        var _a, _b;
        const endpoint = `${this.baseUrl}/scaAssociations`;
        const resource = new resource_1.default(this, endpoint);
        const hasDefinedQueryParams = (_b = (_a = entityType !== null && entityType !== void 0 ? entityType : entityId) !== null && _a !== void 0 ? _a : pageSize) !== null && _b !== void 0 ? _b : pageNumber;
        if (hasDefinedQueryParams) {
            if (!requestOptions)
                requestOptions = {};
            if (!requestOptions.params)
                requestOptions.params = {};
            if (entityType)
                requestOptions.params["entityType"] = entityType;
            if (entityId)
                requestOptions.params["entityId"] = entityId;
            if (pageSize)
                requestOptions.params["pageSize"] = pageSize;
            if (pageNumber)
                requestOptions.params["pageNumber"] = pageNumber;
        }
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "ListAssociationsResponse");
    }
    /**
    * @summary Delete association to devices
    * @param removeAssociationRequest {@link RemoveAssociationRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link void }
    */
    async removeAssociation(removeAssociationRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/scaAssociations`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(removeAssociationRequest, "RemoveAssociationRequest");
        await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "DELETE" });
    }
}
exports.SCAAssociationManagementApi = SCAAssociationManagementApi;
