// @adyen/api-library/lib/src/services/balancePlatform/sCADeviceManagementApi.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SCADeviceManagementApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const objectSerializer_1 = require("../../typings/balancePlatform/objectSerializer");
/**
 * API handler for SCADeviceManagementApi
 */
class SCADeviceManagementApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://balanceplatform-api-test.adyen.com/bcl/v2";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Begin SCA device registration
    * @param beginScaDeviceRegistrationRequest {@link BeginScaDeviceRegistrationRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link BeginScaDeviceRegistrationResponse }
    */
    async beginScaDeviceRegistration(beginScaDeviceRegistrationRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/scaDevices`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(beginScaDeviceRegistrationRequest, "BeginScaDeviceRegistrationRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "BeginScaDeviceRegistrationResponse");
    }
    /**
    * @summary Finish registration process for a SCA device
    * @param deviceId {@link string } The unique identifier of the SCA device that you are associating with a resource.
    * @param finishScaDeviceRegistrationRequest {@link FinishScaDeviceRegistrationRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link FinishScaDeviceRegistrationResponse }
    */
    async finishScaDeviceRegistration(deviceId, finishScaDeviceRegistrationRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/scaDevices/{deviceId}`
            .replace("{" + "deviceId" + "}", encodeURIComponent(String(deviceId)));
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(finishScaDeviceRegistrationRequest, "FinishScaDeviceRegistrationRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "PATCH" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "FinishScaDeviceRegistrationResponse");
    }
    /**
    * @summary Create a new SCA association for a device
    * @param deviceId {@link string } The unique identifier of the SCA device that you are associating with a resource.
    * @param submitScaAssociationRequest {@link SubmitScaAssociationRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link SubmitScaAssociationResponse }
    */
    async submitScaAssociation(deviceId, submitScaAssociationRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/scaDevices/{deviceId}/scaAssociations`
            .replace("{" + "deviceId" + "}", encodeURIComponent(String(deviceId)));
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(submitScaAssociationRequest, "SubmitScaAssociationRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "SubmitScaAssociationResponse");
    }
}
exports.SCADeviceManagementApi = SCADeviceManagementApi;
