// @adyen/api-library/lib/src/services/openBanking/accountVerificationApi.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountVerificationApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const objectSerializer_1 = require("../../typings/openBanking/objectSerializer");
/**
 * API handler for AccountVerificationApi
 */
class AccountVerificationApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://obgateway-test.adyen.com/obgateway/v1";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Create routes for account verification
    * @param accountVerificationRoutesRequest {@link AccountVerificationRoutesRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link AccountVerificationRoutesResponse }
    */
    async createAccountVerificationRoutes(accountVerificationRoutesRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/accountVerification/routes`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(accountVerificationRoutesRequest, "AccountVerificationRoutesRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "AccountVerificationRoutesResponse");
    }
    /**
    * @summary Get account verification report
    * @param code {@link string } The unique code you receive after a successful open banking flow that is included as a query parameter in the &#x60;redirectUrl&#x60; callback.
    * @param requestOptions {@link IRequest.Options }
    * @return {@link AccountVerificationReportResponse }
    */
    async getAccountVerificationReport(code, requestOptions) {
        const endpoint = `${this.baseUrl}/accountVerification/reports/{code}`
            .replace("{" + "code" + "}", encodeURIComponent(String(code)));
        const resource = new resource_1.default(this, endpoint);
        const response = await (0, getJsonResponse_1.default)(resource, "", { ...requestOptions, method: "GET" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "AccountVerificationReportResponse");
    }
}
exports.AccountVerificationApi = AccountVerificationApi;
